### PR TITLE
151 as a dev i want to store inspector notes and comments in fertisca

### PR DESCRIPTION
**Pull request by Francois-Werbrouck**
ref #151

### CODE REVIEW

File: datastore/db/bytebase/FertiScan/schema_0.0.13.sql [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+    "verified" boolean DEFAULT false,
+    "upload_date" timestamp DEFAULT CURRENT_TIMESTAMP,
+    "updated_at" timestamp DEFAULT CURRENT_TIMESTAMP,
+    "inspector_id" uuid NOT NULL REFERENCES "fertiscan_0.0.13".users(id),
+    "label_info_id" uuid REFERENCES "fertiscan_0.0.13".label_information(id),
+    "sample_id" uuid REFERENCES "fertiscan_0.0.13".sample(id),
+    "picture_set_id" uuid REFERENCES "fertiscan_0.0.13".picture_set(id),
+    "user_commment" text,
+    "original_dataset" json
```
Comment (k-allagbe) [inline datastore/db/bytebase/FertiScan/schema_0.0.13.sql:236]: Not sure how `original_dataset` crept in here again but as long as operational transactions are not dependent on it, this is fine by me.

File: tests/fertiscan/inspection_export.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
     "inspection_id": "61c93c2f-78ae-4f90-be3c-fc888087bfb9",
+    "inspection_comment": "",
```
Comment (k-allagbe) [inline tests/fertiscan/inspection_export.json:3]: What's the difference between inspection.json and inspection_export.json? Is it intended that one has `inspection_comment` and not the other?

File: datastore/db/metadata/inspection/__init__.py [hunk 3]
```diff
@@ -347,43 +361,49 @@ def build_inspection_export(cursor, inspection_id, label_info_id) -> str:
...
+        # )
+
+        # Specifications(**(specifications_json.get("specifications")))
 
-        Specifications(**(specifications_json.get("specifications")))
+        # inspection_json.update(specifications_json)è
 
-        inspection_json.update(specifications_json)
+        # Get inspection comment
+        inspection_comment = inspection.get_inspection(cursor,inspection_id)[8]
```
[the remark this answers is not in the document]
Comment (Francois-Werbrouck) [reply datastore/db/metadata/inspection/__init__.py:400]: This is the function to get all the fields from an inspection. As for the magic number you only need to read the documentation of `get_inspection` to understand what it means.

I could create a function to only retrieve the comment, but it seems trivial and brings no value to the project itself. 

I'd rather keep it as it currently is since all the tests also uses `get_inspection` and refer to the right column to verify the right object.

File: datastore/db/bytebase/FertiScan/schema_0.0.13.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (Francois-Werbrouck) [reply datastore/db/bytebase/FertiScan/schema_0.0.13.sql:236]: It doesn't have an impact, but the original_dataset is saved and was merged in a previous issue. It was meant to be there?

File: datastore/db/metadata/inspection/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (k-allagbe) [reply datastore/db/metadata/inspection/__init__.py:400]: I understand. But this is a code smell and pretty easy to avoid. An alternative to using a model is to simply declare a constant variable with an indicative name. Less attractive than a model tbh, because of how many constants would be needed if we wanted to extract the other values.

The documentation doesn't seem to indicate what 8 means.

[the remark this answers is not in the document]
Comment (Francois-Werbrouck) [reply datastore/db/metadata/inspection/__init__.py:400]: Wdym
> An alternative to using a model is to simply declare a constant with an indicative name

I feel like returning a 2D array is standard while querying database rows no?

>he documentation doesn't seem to indicate what 8 means.

I can add a line to specify what the inspection is as follows:
```python
def get_inspection(cursor, inspection_id):
    """
    This function gets the inspection from the database.

    Parameters:
    - cursor (cursor): The cursor of the database.
    - inspection_id (str): The UUID of the inspection.

    Returns:
    - The inspection.
    [ verified, upload_date, updated_at, inspector_id, label_info_id, sample_id, picture_set_id, fertilizer_id, user_comment]
    """

    try:
        query = """
            SELECT 
                verified,
                upload_date,
                updated_at,
                inspector_id,
                label_info_id,
                sample_id,
                picture_set_id,
                fertilizer_id,
                user_comment
            FROM 
                inspection
            WHERE 
                id = %s
            """
...
```

Other than that, I do not know what else I can do to make it more obvious that the 8th index is the user_comment. I could create 8 constant in the module to make it more easy to manage if you'd like. However I've always used array indexes to select a column with a Row in an Array.

[the remark this answers is not in the document]
Comment (k-allagbe) [reply datastore/db/metadata/inspection/__init__.py:400]: Suppose, get_inspection was used in thousand places throughout the code or even accross repos then we change get_inspection is a way that user_comment changes position to 7. This would mean changing 8 to 7 everywhere. With a constant, you would change it only once. With a model, you would change the parsing code only.

But this is not a huge problem. Resolving this.

**Comment by Francois-Werbrouck** [comment]
Tests are passing locally on schema 0.0.15

I will add a test for the edge case of #176 

{BB444A54-2504-43CB-84E7-9B95F1463F69}
